### PR TITLE
file reorg

### PR DESCRIPTION
**Pull request by evykassirer**
TODO:

- [x] move tests for stuff in the util folder into the util folder
- [x] add more stuff into util folder: remove parens, flatten, possibly more
- [x] move rearrange coeff into simplify basics or a top level tree search
- [x] break up simplifyBasics functions so each one is its own file and tested by a separate file
- [x] split simplifyExpression into simplify file and stepThrough file (don’t export `step` at all, use the first step from stepThrough for tests that test `step`)

... and anything else!

NOT IN THIS PR: moving removeParens, that should be it's own PR

### CODE REVIEW

**Comment by evykassirer** [comment]
I'm gonna work on this some more today probably

File: index.js [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
-const solveEquation = require('./lib/solveEquation');
+const simplifyExpressionNode = require('./lib/simplifyExpressionNode');
+const simplifyExpressionString = require('./lib/simplifyExpressionString');
+const solveEquation = require('./lib/solveEquation/solveEquation');
+const solveEquationString = require('./lib/solveEquation/solveEquationString');
+const MathChangeTypes = require('./lib/MathChangeTypes');
 
 module.exports = {
-  expressionStepper: simplifyExpression.stepThrough,
+  // TODO: after deploying this and then the update to Athena.
```
Comment (evykassirer) [inline index.js]: this part is important to see

File: lib/PolynomialTermOperations.js [hunk 2]
```diff
@@ -1,420 +0,0 @@
-'use strict';
```
Comment (evykassirer) [inline lib/PolynomialTermOperations.js]: PolynomialTermOperations.js is gone now! :)

File: lib/collectAndCombine/multiplyLikeTerms.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  status = multiplyPolynomialTerms(node);
+  if (status.hasChanged()) {
+    status.changeType = MathChangeTypes.MULTIPLY_COEFFICIENTS;
+    return status;
+  }
+
+  return NodeStatus.noChange(node);
+}
+
+function multiplyPolynomialTerms(node) { // TODO: do this for add too
```
Comment (evykassirer) [inline lib/collectAndCombine/multiplyLikeTerms.js]: oh I can take out this TODO

File: lib/evaluateFunctions/nthRoot.js [hunk 4]
```diff
@@ -2,70 +2,16 @@
...
-      math.abs(argument.args[1].eval()));
-    newNode = NodeCreator.operator('/', [newNumerator, newDenominator]);
-    return NodeStatus.nodeChanged(
-      MathChangeTypes.ABSOLUTE_VALUE, node, newNode);
-  }
-  else {
-    return NodeStatus.noChange(node);
-  }
-}
+// TODO(ael): make nthRoot a folder and sort out these functions better
```
Comment (evykassirer) [inline lib/evaluateFunctions/nthRoot.js]: left this for later, cause I wasn't sure what the best way to structure it was

File: lib/simplifyBasics/simplifyBasicsSearch.js [hunk 5]
```diff
@@ -5,15 +5,18 @@
...
+const Negative = require('../util/Negative');
+const NodeCreator = require('../util/NodeCreator');
+const NodeStatus = require('../util/NodeStatus');
+const NodeType = require('../util/NodeType');
+const TreeSearch = require('../util/TreeSearch');
+
+const reduceMultiplicationByZero = require('./reduceMultiplicationByZero');
+const reduceZeroDividedByAnything = require('./reduceZeroDividedByAnything');
+const rearrangeCoefficient = require('./rearrangeCoefficient');
+// TODO: move the rest of the functions into their own files too
```
Comment (evykassirer) [inline lib/simplifyBasics/simplifyBasicsSearch.js]: I might make this an issue, it's not super important but would feel cleaner I think

**Comment by aelnaiem** [comment]
Let's merge this :)

**Comment by evykassirer** [comment]
k one sec gonna remove a stale comment and merge

**Pull request merged**